core/common/test/samples/TimeZoneSamples.kt

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2019-2024 JetBrains s.r.o. and contributors.
+ * Use of this source code is governed by the Apache 2.0 License that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.datetime.test.samples
+
+import kotlinx.datetime.*
+import kotlinx.datetime.format.*
+import kotlin.test.*
+
+class TimeZoneSamples {
+
+    @Test
+    fun usage() {
+        val zone = TimeZone.of("Europe/Berlin")
+        val localDateTime = LocalDate(2021, 3, 28).atTime(2, 16, 20)
+        val instant = localDateTime.toInstant(zone)
+        check(instant == Instant.parse("2021-03-28T01:16:20Z"))
+        val newLocalDateTime = instant.toLocalDateTime(zone)
+        check(newLocalDateTime == LocalDate(2021, 3, 28).atTime(3, 16, 20))
+    }
+
+    @Test
+    fun id() {
+        check(TimeZone.of("America/New_York").id == "America/New_York")
+    }
+
+    @Test
+    fun toStringSample() {
+        val zone = TimeZone.of("America/New_York")
+        check(zone.toString() == "America/New_York")
+        check(zone.toString() == zone.id)
+    }
+
+    @Test
+    fun equalsSample() {
+        val zone1 = TimeZone.of("America/New_York")
+        val zone2 = TimeZone.of("America/New_York")
+        check(zone1 == zone2) // different instances, but the same ID
+        val zone3 = TimeZone.of("UTC+01:00")
+        val zone4 = TimeZone.of("GMT+01:00")
+        check(zone3 != zone4) // the same time zone rules, but different IDs
+    }
+
+    @Test
+    fun currentSystemDefault() {
+        // a fixed-width format for log entries
```

Comment:
This example neither prints anything (e.g. meaningless as a runnable sample) nor gives a taste what to expect.

Consider replacing the last call with something like
```
// [2024-04-16 15:05:58.014+0200] Starting the application
println(logEntry("Starting the application"))
```